src/Monomer/Graphics/Types.hs

```diff
@@ -45,11 +45,16 @@ data Color = Color {
 instance Default Color where
   def = Color 255 255 255 1.0
 
--- | The definition of a font.
-data FontDef = FontDef {
-  _fntName :: !Text,  -- ^ The logic name. Will be used when defining styles.
-  _fntPath :: !Text   -- ^ The path in the filesystem.
-} deriving (Eq, Show, Generic)
+data FontDef
+  = FontDefFile
+    { _fntFontName :: !Text  -- ^ The logic name. Will be used when defining styles.
+    , _fntFontPath :: !Text  -- ^ The path in the filesystem.
+    }
+  | FontDefMem
+    { _fntFontName :: !Text         -- ^ The logic name. Will be used when defining styles.
+    , _fntFontBytes :: !ByteString  -- ^ The bytes of the loaded font.
+    }
+  deriving (Eq, Show, Generic)
```

Comment:
Also note this change creates traversals, not lenses, which is something to watch out for, see GHCI:

```haskell
λ: FontDefMem "name" "Bytes" ^. fontPath 
""
```

I'm open to alternatives, but given how small these constructors are, I'm fine with leaving as is.

Reply:
I'm fine with the naming you chose. I guess the error is caused by a previous Lens/Prism generated in https://github.com/fjvallarino/monomer/blob/main/src/Monomer/Lens.hs. 

---

Alternatively, since this constructors are only used in a couple of places, you can use:

```haskell
data FontDef
  = FontDefFile !Text !Text -- ^ Associates a font name with a filesystem path.
  | FontDefMem !Text !ByteString -- ^ Associates a font name with a font loaded in memory.
  deriving (Eq, Show, Generic)
```

Then, in `NanoVGRenderer`, you need to deconstruct:

```haskell
...
    createFont (FontDefFile name path) = VG.createFont c name (VG.FileName path)
    createFont (FontDefMem name bytes) = VG.createFontMem c name bytes
...
```

Reply:
Thanks, I went with what you suggested. `name` was needed outside of the `createFont` function in both `NanoVGRenderer` and `FontManager`, so I hand-crafted a `HasName` instance for `FontDef`, since it can't be generated anymore due to no record syntax.